README.md

```diff
@@ -0,0 +1,151 @@
+
+- [Introduction](#introduction)
+- [Innovation](#innovation)
+- [How to Use](#how-to-use)
+- [Intermediate Files Generated](#intermediate-files-generated)
+- [Operation Stages of the Tool](#operation-stages-of-the-tool)
+  - [I. Compilation File List Generation](#i-compilation-file-list-generation)
+  - [II. Git Diff Report Generation](#ii-git-diff-report-generation)
+  - [III. Commit Metadata Retrieval](#iii-commit-metadata-retrieval)
+  - [IV. Web Script Generation](#iv-web-script-generation)
+
+## Introduction
+
+DeltaKernel Change Impact Analysis Tool generates a visual report detailing changes in both header files and source code between two Linux kernel versions (tags in the Linux kernel repository: old_tag and new_tag). This tool helps developers compare updates between versions.
+
+The diff report includes a subset of files from the Linux kernel repository that are included in building the kernel, contributing to a focused and detailed report on the compile-time source code in Linux.
+
+## Innovation
+
+The idea of generating a web display for Linux kernel version change impact analysis is inspired by [Cregit](https://github.com/cregit/cregit). This tool improves on Cregit by:
+
+- The tool performs a version analysis between two tags to identify updated files, similar to git diff, but focuses only on files used to compile the kernel rather than the entire Linux source code.
+
+- Generating not only web source files but also lists of all source files and dependencies/header files used in kernel compilation, facilitating additional analysis purposes. (More details in [Intermediate Files Generated](#intermediate-files-generated))
+- Enabling comparison between two specific tags/releases in the Linux kernel, highlighting all newly added and deleted lines. This provides a clear layout of differences between the tags. While Cregit organizes information by files and embeds the latest commit details in each line/token, it does not support direct comparison between two tags.
+- User customization: allows users to define the URL of the Linux kernel repository and specify the specific subsystem for analysis. (More details in [How to Use](#how-to-use))
+- Kernel Configuration: The linux kernel is configured with `make olddefconfig` in `build_scripts/build_collect_diff`: Updates the configuration using the existing `.config` file and applies default values for new options.
+
+## How to use
+
+To utilize this tool in your Linux environment (tested successfully on `Ubuntu 22.04`), follow these steps:
+
+**Clone the repository**:
+
+```bash
+git clone <repository_url>
+```
+
+**Navigate to the cloned repository**:
+
+```bash
+cd <repository_directory>
+```
+
+**Execute the tool by specifying the old and new tags**:
+
+```bash
+./run_tool <tag1> <tag2> [-c clone_path] [-u repo_link] [-s subsystem]
+```
+
+**Example Usage**:
+
+```bash
+./run_tool "v5.15" "v5.15.100" -c "linux-clone" -u "https://github.com/torvalds/linux" -s "security"
```

Comment:
It looks like your tags got out of sync with your comment. 

```suggestion
./run_tool "v6.8" "v6.9" -c "linux-clone" -u "https://github.com/torvalds/linux" -s "security"
```

Reply:
Actually I have updated all to v5.15 vs v5.15.100, for later version might run into conflict issue when applying the patch.